### PR TITLE
Difficulty Selection Improvements

### PR DESCRIPTION
**Pull request by Kevbyte**

### CODE REVIEW

**Comment by Kevbyte** [comment]
LGTM

**Comment by codydaig** [comment]
Rebash, squash, and I'll give the 4 letters.

**Comment by AndrewKishino** [comment]
lgtm

File: client/app/profile/profile.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    // a notification should pop up on both screens
+    $scope.userHasBattleRequest = true;
+    $scope.battleRequestStatus = 'init';
+    $scope.$apply();
+    
+    // the url hash needs to also be sent to the player who accepted the
+    // challenge
+  });
+  
+});
```
Comment (codydaig) [inline client/app/profile/profile.js]: New line

File: client/app/profile/profile.html [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            </ul>
+          </div><!-- /dashboar widget -->
+        </div><!-- /grid item -->
+      </div><!-- /grid -->
+
+    </div><!-- dashboard -->
+  </div><!-- /container -->
+</main><!-- /main -->
+
+<footer-partial></footer-partial>
```
Comment (codydaig) [inline client/app/profile/profile.html]: new line

File: client/app/profile/profile.js [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+  ////////////////////////////////////////////////////////////
+
+  $scope.getLeaderboard = function(username) {
+    // $scope.leaderboard = Users.getLeaderboard();
+    Users.getLeaderboard()
+      .then(function(leaderboard){ 
+        $scope.leaderboard = leaderboard.data;
+        console.log($scope.leaderboard);
+      });
+  }
```
Comment (codydaig) [inline client/app/profile/profile.js]: semicolon

File: client/app/profile/profile.js [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+  $scope.getStats = function(username) {
+    Users.getStats(username)
+      .then(function(stats){
+        $scope.currentStreak = stats.currentStreak;
+        $scope.longestStreak = stats.longestStreak;
+        $scope.totalWins = stats.totalWins;
+        $scope.points = $scope.totalWins * 10;
+      });
+  }
```
Comment (codydaig) [inline client/app/profile/profile.js]: semicolon

File: client/app/profile/profile.js [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+
+  ////////////////////////////////////////////////////////////
+  // handle battle requests
+  ////////////////////////////////////////////////////////////
+
+  // Open up socket with specific dashboard server handler
+  $scope.requestBattle = function($event, opponentUsername) {
+    $event.preventDefault();
+    if (!$scope.challengeLevel) $scope.challengeLevel = 4;
+    console.log("CHALLENGE LEVEL: ", $scope.challengeLevel);
```
Comment (codydaig) [inline client/app/profile/profile.js]: Comment Out

File: client/app/profile/profile.js [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+  };
+
+  // listen for incoming battle request
+  $rootScope.dashboardSocket.on('incomingBattleRequest', function(userData){
+    $scope.battleRequestOpponentName = userData.fromUser;
+    $scope.battleRequestChallengeLevel = userData.challengeLevel;
+    $scope.userHasBattleRequest = true;
+    $scope.battleRequestStatus = 'open';
+    $scope.$apply();
+    console.log("opponent has challenged you: ", userData.fromUser);
```
Comment (codydaig) [inline client/app/profile/profile.js]: Comment out

File: client/app/profile/profile.js [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+  });
+
+  // battle has been accepted
+  $scope.battleAccepted = function() {
+    console.log("CHALLENGE ACCEPTED, CHALLENGE LEVEL: ", $scope.battleRequestChallengeLevel);
```
Comment (codydaig) [inline client/app/profile/profile.js]: comment out

File: client/app/profile/profile.js [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+    $scope.battleRequestStatus = 'none';
+  };
+
+  // prepare for battle, only gets fired when a user has sent a battle request,
+  // and another user has accepted.
+  $rootScope.dashboardSocket.on('prepareForBattle', function(data) {
+    // at this point, the opponent (i.e. the person who sent the initial battle
+    // request) should be notified that the person he/she challenged has
+    // accepted.
+    console.log('prepare for battle!', data);
```
Comment (codydaig) [inline client/app/profile/profile.js]: comment out

File: client/app/profile/profile.js [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+  ////////////////////////////////////////////////////////////
+  // get user stats for leaderboard
+  ////////////////////////////////////////////////////////////
+
+  $scope.getLeaderboard = function(username) {
+    // $scope.leaderboard = Users.getLeaderboard();
+    Users.getLeaderboard()
+      .then(function(leaderboard){ 
+        $scope.leaderboard = leaderboard.data;
+        console.log($scope.leaderboard);
```
Comment (codydaig) [inline client/app/profile/profile.js]: comment out

File: client/app/profile/profile.js [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+  ////////////////////////////////////////////////////////////
+
+  $scope.onlineUsers;
+
+
+  $rootScope.dashboardSocket.on('updateUsers', function(data) {
+    //TODO: Online users.
+
+    console.log('NEED TO UPDATE USERS CUZ OF SOME EVENT');
+    console.log(data);
```
Comment (codydaig) [inline client/app/profile/profile.js]: comment these out

File: client/app/dashboard/dashboard.js [hunk 11]
```diff
@@ -97,7 +115,7 @@ angular.module('battlescript.dashboard', [])
   // Open up socket with specific dashboard server handler
   $scope.requestBattle = function($event, opponentUsername) {
     $event.preventDefault();
-    if (!$scope.challengeLevel) $scope.challengeLevel = 4;
+    if (!$scope.challengeLevel) $scope.challengeLevel = 4 + Math.floor(Math.random()*2);
     console.log("CHALLENGE LEVEL: ", $scope.challengeLevel);
```
Comment (codydaig) [inline client/app/dashboard/dashboard.js]: comment out

File: client/app/dashboard/dashboard.js [hunk 12]
```diff
@@ -86,7 +104,7 @@ angular.module('battlescript.dashboard', [])
         $scope.leaderboard = leaderboard.data;
         console.log($scope.leaderboard);
```
Comment (codydaig) [inline client/app/dashboard/dashboard.js]: comment out

**Comment by codydaig** [comment]
LGTM
@linalu1 :thumbsup:

**Pull request merged**